relay-server/src/services/store.rs

```diff
@@ -896,13 +902,52 @@ impl StoreService {
                     false
                 }
             });
-        }
 
-        span.duration_ms = ((span.end_timestamp - span.start_timestamp) * 1e3) as u32;
-        span.event_id = event_id;
-        span.project_id = scoping.project_id.value();
-        span.retention_days = retention_days;
-        span.start_timestamp_ms = (span.start_timestamp * 1e3) as u64;
+            let group = span
+                .sentry_tags
+                .as_ref()
+                .and_then(|sentry_tags| sentry_tags.get("group").cloned())
+                .unwrap_or_default();
+
+            for (mri, summaries) in metrics_summary {
+                let Some(summaries) = summaries else {
+                    continue;
+                };
+                for summary in summaries {
+                    let Some(summary) = summary else {
+                        continue;
+                    };
+                    // Ignore immediate errors on produce.
+                    if let Err(error) = self.produce(
+                        KafkaTopic::MetricsSummaries,
+                        scoping.organization_id,
+                        KafkaMessage::MetricsSummary(MetricsSummaryKafkaMessage {
+                            count: summary.count,
+                            duration_ms: span.duration_ms,
+                            end_timestamp: span.end_timestamp,
+                            group: &group,
+                            is_segment: span.is_segment,
+                            max: summary.max,
+                            mri,
+                            min: summary.min,
+                            project_id: span.project_id,
+                            retention_days: span.retention_days,
+                            segment_id: span.segment_id.unwrap_or_default(),
+                            span_id: span.span_id,
+                            sum: summary.sum,
+                            tags: summary.tags,
+                            trace_id: span.trace_id,
+                        }),
+                    ) {
+                        relay_log::error!(
+                            error = &error as &dyn std::error::Error,
+                            "failed to push metrics summary to kafka",
+                        );
+                    }
+                }
+            }
+            span.metrics_summary = None;
```

Comment:
`SpanKafkaMessage` should document which fields are actually sent to Kafka on the `spans` topic, so I suggest we remove this field from the type and deserialize into a separate helper struct instead.

```rust
[derive(Deserialize)]
struct Helper {
    metrics_summary: Option<BTreeMap<Cow<'a, str>, Option<SpanMetricsSummaries<'a>>>>
}